crates/node-types/src/lib.rs

```diff
@@ -85,6 +100,40 @@ impl Default for BigBang {
     }
 }
 
+#[cfg(feature = "new-block")]
+impl BlockTx {
+    /// Create a new [`BlockTx`] to notify listeners of new blocks.
+    pub fn new() -> Self {
+        let (block_tx, _block_rx) = tokio::sync::watch::channel(());
+        Self(block_tx)
+    }
+
+    /// Notify listeners that a new block has been received.
+    ///
+    /// Note this is best effort and will still send even if there are currently no listeners.
+    pub fn notify(&self) {
+        let _ = self.0.send(());
+    }
+
+    /// Create a new [`BlockRx`] to listen for new blocks.
+    pub fn new_listener(&self) -> BlockRx {
+        BlockRx(self.0.subscribe())
+    }
+
+    /// Get the number of receivers listening for new blocks.
+    pub fn receiver_count(&self) -> usize {
+        self.0.receiver_count()
+    }
+}
+
+#[cfg(feature = "new-block")]
+impl BlockRx {
+    /// Waits for a change notification.
+    pub async fn changed(&mut self) -> Result<(), tokio::sync::watch::error::RecvError> {
+        self.0.changed().await
+    }
+}
+
```

Comment:
Maybe we can move the `BlockTx` and `BlockRx` types and their impls into a `block_notify` module or similar? Then we can just have one `#[cfg(feature` over the module itself?